### ISSUE TITLE
Toolkit stopped listing profiles with RoleArn and EndpointName

### ISSUE THREAD

**Issue by arunr21487** [aws/aws-toolkit-visual-studio#156]
**Describe the bug**

- We are using Visual Studio 2019 Community Edition for our application development. We are developing an application in .Net, in C# language. 

- Presently the AWS Explorer is not listing the Profiles based on the Roles which was working correctly previously

**To Reproduce**

- In Visual Studio, Go to Extensions -> Manage Extensions. Then click on Online tab and then download and install **AWS Toolkit for Visual Studio 2017 and 2019**
- Then go to the search bar of Visual Studio and enter "**Developer PowerShell**" and select that
- Then execute the below PowerShell script:
- $endpoint=_aws endpoint_

$cred = Get-Credential -Message "Enter domain user (just username) and password for SAML AWS authentication"

Write-Output "Requesting SAML token for user $cred.Username against role $role in region $region"

Set-AWSSamlEndpoint -Endpoint $endpoint -StoreAs 'endpoint'

Set-AWSSamlRoleProfile -EndpointName "endpoint" -StoreAllRoles

- When we execute this script, Windows PowerShell credentials will be requested and enter the correct credentials
- When we enter the credentials correctly, after executing the PowerShell scripts, we can see that the AWS Role Profiles assigned to that credentials will be listed in the PowerShell window
- After that, restart the Visual Studio and go to View -> AWS Explorer
- Previously, when we go to AWS Explorer after executing above steps, AWS Role Profiles were listing in the drop down list. But now we can see that the **Credentials** drop down list is empty. Previously, the name of the drop down list was something like **Profiles**
- When we go to View -> Output -> Amazon Web Services, we can see like **Profiles found: 2. The following credentials could not be loaded: _Profile names_ . Check the Toolkit logs for more details**
- When we go to AWS Toolkit logs(%USERPROFILE%\AppData\Local\AWSToolkit\logs\visualstudio), we can see an error '**missing required property SourceProfile**'
- Please find below error we are getting in AWS Toolkit logs(%USERPROFILE%\AppData\Local\AWSToolkit\logs\visualstudio):

- 2021-05-07 19:55:06,608 [31] ERROR Amazon.AWSToolkit.Credentials.Core.ProfileCredentialProviderFactory - The following credentials could not be loaded: 
1111111:role/_role_-role: Assume Role Profile _profilename_ is missing required property SourceProfile
1111111:role/_role_-role: Assume Role Profile _profilename_ is missing required property SourceProfile
- This issue is there in latest version(1.21.1.0). When we check in 1.18.x.x version, this is working correctly

- Kindly let us know if you need any further information
- Kindly help us on this at the earliest as we are completely stuck due to this issue

**Computer (please complete the following information):**

- Windows Version: Windows 10
- Visual Studio Version: Visual Studio 2019 Community Edition
- AWS Toolkit for Visual Studio Version: 1.21.1.0

**Additional context**
- This issue is there in latest version(1.21.1.0). When we check in 1.18.x.x version, this is working correctly

**Comment by awschristou** [comment]
Thank you for reaching out and including a detailed set of repro steps @arunr21487 - We will investigate this. 

I don't have a workaround for you at this time, however the new version of the Toolkit has support for AWS SSO based credentials (here is the [user documentation](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/sso-credentials.html)). Have you tried integrating your SAML provider into AWS SSO? You might be able to use the Toolkit without requiring those PowerShell steps to manually gain credentials.

**Comment by arunr21487** [comment]
Hi..Any updates on this?
We also tried the method you suggested in the documentation you shared, using AWS SSO based credentials, but we are getting the following error:
ERROR Amazon.AWSToolkit.Credentials.Core.AwsConnectionManager - Failed to switch to credentials: Profile:sso-user-1
Amazon.SSOOIDC.Model.InvalidRequestException: The service returned an error with Error Code InvalidRequestException and HTTP Body: {"error":"invalid_request","error_description":"Invalid request"} ---> Amazon.Runtime.Internal.HttpErrorResponseException: The remote server returned an error: (400) Bad Request. ---> System.Net.WebException: The remote server returned an error: (400) Bad Request.

Actually our most of the developers are using method using PowerShell scripts and it was working correctly before. Kindly help us on that.

Or kindly provide executable of your bit older version like 1.18.x.x version which was working. We were not able get that from the MarketPlace.

**Comment by awschristou** [comment]
Hi @arunr21487, we have created a test build of the toolkit that should support credentials profiles containing role arn and endpointname properties. We'd appreciate if you could try it out and let us know if this unblocks you.

[AWSToolkitPackage.zip](https://github.com/aws/aws-toolkit-visual-studio/files/6497216/AWSToolkitPackage.zip)
sha384: `4f645e8fb6ed1804729c65f496f6f29c54514e7381633b4837c03facb1c62b15de84d70c75f2c01a3b3ffda01ebb666f`

**Comment by arunr21487** [comment]
Hi @awschristou, We tested your test build and we could see that our issue is fixed in this build. Thanks for the support.

**Comment by arunr21487** [comment]
Hi @awschristou, We will tell the confirmation after we complete one round of end to end testing

**Comment by awschristou** [comment]
Thank you for taking a look @arunr21487 ! Unless I hear otherwise I will assume your two messages above are out of order and this fix will unblock your situation.

**Comment by arunr21487** [comment]
Hi @awschristou These messages are not out order. When we did a random testing, we could see that the issue is fixed. But we need to do one round of end to end testing and confirm the same. After this, we will get back to you at the earliest.

**Comment by arunr21487** [comment]
Hi @awschristou, We completed our end to end testing and confirmed that the AWS Toolkit for Visual Studio issue is fixed in this test build. Kindly let us know the timeline for the release of this version as our development team is fully stuck with this issue. Also we want to let you know that are facing the same issue in AWS Toolkit for Visual Studio Code latest version as well. Kindly do the necessary actions to fix the issue there as well as we think it is really unnecessary to raise another separate ticket for the same.

**Comment by awschristou** [comment]
Thank you for your patience and for helping verify the fix @arunr21487 

We have released version 1.21.2.0 of the Toolkit today, which now shows profiles from the SDK Credentials Store that contain the "RoleArn" and "EndpointName" properties.

For assistance with the AWS Toolkit for VS Code, you will have to file an issue at https://github.com/aws/aws-toolkit-vscode/issues . Please include details around whether this was working in a previous version and stopped, or if this is a new feature request for that Toolkit, and an example credentials entry (illustrating the properties used, but without including any account specific details).
